### PR TITLE
Top menu

### PR DESCRIPTION
**Pull request by CoyotWilly**
## What's new 🆕 
- Top menu 
- Observable coord system - selectable menu
- Sliding for  top panel
- Display of currently chosen `$BASE` and `$TOOL`
- Docker file

## From lab 🥼 🧪 
- Some bug fixes for Java server
- Bug fix of Unity internal coord system
- General code imporvement
- #49 fix

## And much more 🧾 
- Some UI documentation
- New  assets

### CODE REVIEW

File: unity/kuka-ar-unity/Assets/Project/Scripts/EventSystem/Controllers/Menu/TopMenuController.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        public int transformFactor;
+        
+        [Tooltip("Menu drop height, counted from screen height [%]")]
+        public float dropScreenHeight;
+        
+        [Tooltip("Jogs component reference object")]
+        public GameObject jogs;
+
+        [Tooltip("Self reference to prevent null pointer exception on behavior")]
+        public GameObject topMenu;
```
Comment (xalpol12) [inline unity/kuka-ar-unity/Assets/Project/Scripts/EventSystem/Controllers/Menu/TopMenuController.cs:26]: 🤔 why no [Unity gameobject docs](https://docs.unity3d.com/ScriptReference/Component-gameObject.html)

**Comment by xalpol12** [comment]
@CoyotWilly can you look at #[3e5fe75](https://github.com/xalpol12/kuka-ar-all/pull/46/commits/3e5fe7581f39bcbb04dac1b07d13a1c7a74c2853) and provide feedback? My people said it's more optimal to allocate it using Stack

File: unity/kuka-ar-unity/Assets/Project/Scripts/EventSystem/Controllers/Menu/TopMenuController.cs (hunk 1 above)
Comment (CoyotWilly) [reply unity/kuka-ar-unity/Assets/Project/Scripts/EventSystem/Controllers/Menu/TopMenuController.cs:26]: I'm gonna check this one, but if my memory is correct where was some problem with it.

**Comment by CoyotWilly** [comment]
> @CoyotWilly can you look at #[3e5fe75](https://github.com/xalpol12/kuka-ar-all/pull/46/commits/3e5fe7581f39bcbb04dac1b07d13a1c7a74c2853) and provide feedback? My people said it's more optimal to allocate it using Stack

Your ppl are right, it works much better.

File: unity/kuka-ar-unity/Assets/Project/Scripts/EventSystem/Controllers/Menu/TopMenuController.cs (hunk 1 above)
Comment (CoyotWilly) [reply unity/kuka-ar-unity/Assets/Project/Scripts/EventSystem/Controllers/Menu/TopMenuController.cs]: I modified this one, in dc5754206ceb4c513c8ec437408bef97fea77482 if you could check is it fine. There was small problem but i solved this somewhere during implementation of TOP MENU, so this suggestion works as it should now.

File: unity/kuka-ar-unity/Assets/Project/Scripts/EventSystem/Controllers/Menu/TopMenuController.cs [hunk 2]
```diff
@@ -10,32 +7,13 @@ namespace Project.Scripts.EventSystem.Controllers.Menu
...
-
         [SerializeField]
         [Tooltip("Tracked robots handler reference game object")]
         private GameObject robotModel;
 
         private TrackedRobotsHandler trackedRobot;
-        private TMP_Text toolNo;
-        private TMP_Text baseNo;
+        private TMP_Text tool;
+        private TMP_Text world;
```
Comment (xalpol12) [inline unity/kuka-ar-unity/Assets/Project/Scripts/EventSystem/Controllers/Menu/TopMenuController.cs:16]: It should be base, right now we track the tool and base number.

File: unity/kuka-ar-unity/Assets/Project/Scripts/EventSystem/Controllers/Menu/TopMenuController.cs [hunk 3]
```diff
@@ -44,57 +22,29 @@ private void Awake()
...
             var constantTopPanel = topMenu.GetComponent<RectTransform>()
                 .Find("ConstantInfoPanel").GetComponent<RectTransform>().gameObject;
             
-            toolNo = constantTopPanel.transform.Find("ToolBean").GetComponent<RectTransform>().gameObject
+            tool = constantTopPanel.transform.Find("ToolBean").GetComponent<RectTransform>().gameObject
+                .transform.Find("ToolBackground").GetComponent<RectTransform>().gameObject
                 .transform.Find("Tool").GetComponent<TMP_Text>();
-            baseNo = constantTopPanel.transform.Find("WorldBean").GetComponent<RectTransform>().gameObject
+            world = constantTopPanel.transform.Find("WorldBean").GetComponent<RectTransform>().gameObject
+                .transform.Find("WorldBackground").GetComponent<RectTransform>().gameObject
```
Comment (xalpol12) [inline unity/kuka-ar-unity/Assets/Project/Scripts/EventSystem/Controllers/Menu/TopMenuController.cs:33]: same idea here, "base" not "world"

File: unity/kuka-ar-unity/Assets/Project/Scripts/EventSystem/Controllers/Menu/TopMenuController.cs [hunk 4]
```diff
@@ -44,57 +22,29 @@ private void Awake()
...
-            if (id == uid)
-            {
-                CoordListState = LogicStates.AutoPulling;
-            }
-        }
-
         private void OnBaseValueChange(object _, KRLInt e)
         {
-            baseNo.text = "Base\n" + e.Value;
+            world.text = "World\n" + e.Value;
```
Comment (xalpol12) [inline unity/kuka-ar-unity/Assets/Project/Scripts/EventSystem/Controllers/Menu/TopMenuController.cs:42]: same here

Comment (CoyotWilly) [reply unity/kuka-ar-unity/Assets/Project/Scripts/EventSystem/Controllers/Menu/TopMenuController.cs]: ## After review fix 
All of this point has been fix in 82441b478fb91e017afd338742a38d835f7da170

**Review by xalpol12 — APPROVED** [review]
Looks good, optimised UI, now we can select base or tool or both. I will implement the logic for these buttons in the weekend and also try to diagnose and fix #47 and #48

**Comment by CoyotWilly** [comment]
As soon as I'm done with SCADA tasks for this week, I'll move all my brain cells to work over #49 and #49, unfortunately I'm gonna find out what do I have to do tommorow 😐 .

**Comment by xalpol12** [comment]
Sure, I will be in contact with you and @przWaw to consult those issues when needed 👍

**Pull request merged**